src/traces/formatTraces.ts

```diff
@@ -46,88 +46,99 @@ export const createListTracesDataFrame = (
-      { name: 'Error Count', type: FieldType.number, values: errors },
-      { name: 'Last Updated', type: FieldType.time, values: lastUpdated },
-    ],
-  };
-  const dataFrames = new MutableDataFrame(traceFields);
-  return { data: [dataFrames], key: targets[0].refId };
+        { name: 'Trace Group', type: FieldType.string, values: traceGroups },
+        { name: 'Latency (ms)', type: FieldType.number, values: latency },
+        // TODO: { name: 'Percentile in trace group', type: FieldType.string, values: ['todo'] },
+        { name: 'Error Count', type: FieldType.number, values: errors },
+        { name: 'Last Updated', type: FieldType.time, values: lastUpdated },
+      ],
+    };
+    return new MutableDataFrame(traceFields);
+  }
+  // if multiple targets of type traceList, map them into data
+  const dataFrames = response.map(res => createDataFrame(res));
+
+  return { data: dataFrames, key: targets[0].refId };
 };
 
-export const createTraceDataFrame = (targets, response: OpenSearchSpan[]): DataQueryResponse => {
-  // first, transform Open Search response to fields Grafana Trace View plugin understands
-  const spans = transformTraceResponse(response);
-
-  const spanFields = [
-    'traceID',
-    'serviceName',
-    'parentSpanID',
-    'spanID',
-    'operationName',
-    'startTime',
-    'duration',
-    'tags',
-    'serviceTags',
-    'stackTraces',
-    'logs',
-  ];
-
-  let series = createEmptyDataFrame(spanFields, '', false, QueryType.Lucene);
-  const dataFrames: DataFrame[] = [];
-  // Add a row for each document
-  for (const doc of spans) {
-    series.add(doc);
+export const createTraceDataFrame = (
+  targets,
+  response: { responses: Array<{ hits: { hits: OpenSearchSpan[] } }> }
+): DataQueryResponse => {
+  function getDataFrameForTarget(target, response) {
+    // first, transform Open Search response to fields Grafana Trace View plugin understands
+    const spans = transformTraceResponse(response);
+
+    const spanFields = [
+      'traceID',
+      'serviceName',
+      'parentSpanID',
+      'spanID',
+      'operationName',
+      'startTime',
+      'duration',
+      'tags',
+      'serviceTags',
+      'stackTraces',
+      'logs',
+    ];
+
+    let series = createEmptyDataFrame(spanFields, '', false, QueryType.Lucene);
+    const dataFrames: DataFrame[] = [];
+    // Add a row for each document
+    for (const doc of spans) {
+      series.add(doc);
+    }
+    series.refId = target.refId;
+    series = addPreferredVisualisationType(series, 'trace');
+    dataFrames.push(series);
+    return dataFrames;
   }
-  // do we need this?
-  series.refId = targets[0].refId;
-  series = addPreferredVisualisationType(series, 'trace');
-  dataFrames.push(series);
+  // if multiple targets of type: trace, flatMap them into data(should be a single array with all trace data frames)
+  const data = targets.flatMap((target, index) => getDataFrameForTarget(target, response.responses[index].hits.hits));
```

Comment:
since queries come in the same order that they're [sent,](https://opensearch.org/docs/latest/api-reference/multi-search/#response) I think we can count on the index of target to be the same as index of response for that target